llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUInstPrinter.cpp

```diff
@@ -454,6 +454,8 @@ void AMDGPUInstPrinter::printImmediate64(uint64_t Imm,
   else if (Imm == 0x3fc45f306dc9c882 &&
            STI.hasFeature(AMDGPU::FeatureInv2PiInlineImm))
     O << "0.15915494309189532";
+  else if (IsFP && AMDGPU::isValid32BitLiteral(Imm, true))
```

Comment:
Can this be:
```suggestion
  else if (IsFP) {
    assert(AMDGPU::isValid32BitLiteral(Imm, true));
```

Reply:
Yes, it can, given that lo32 are cleared by the parser. Changed.